graphqlapi/src/main/java/io/stargate/graphql/schema/types/TupleBuilder.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.types;
+
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLInputType;
+import graphql.schema.GraphQLNonNull;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLOutputType;
+import graphql.schema.GraphQLType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TupleBuilder extends TypeBuilder {
+  private final List<GraphQLType> subTypes;
+
+  public TupleBuilder(List<GraphQLType> subTypes) {
+    this.subTypes = subTypes;
+  }
+
+  private String getName(boolean isInput) {
+    return String.format(
+        "Tuple%s%s",
+        subTypes.stream().map(this::getTypeName).collect(Collectors.joining()),
+        isInput ? "Input" : "");
+  }
+
+  public GraphQLOutputType buildOutputType() {
+    List<GraphQLFieldDefinition> fields = new ArrayList<>(subTypes.size());
+
+    for (int i = 0; i < subTypes.size(); i++) {
+      GraphQLOutputType t = (GraphQLOutputType) subTypes.get(i);
+      fields.add(
+          GraphQLFieldDefinition.newFieldDefinition()
+              .name(String.format("item%d", i))
+              .type(i == 0 ? new GraphQLNonNull(t) : t)
```

Comment:
I think CQL allows all fields to be null:
```
> create table foo(k int primary key, t tuple<int,int>);
> insert into foo(k,t) values (1,(null,null));
> select * from foo;

 k | t
---+--------------
 1 | (None, None)

```

Reply:
TIL, nice catch!

I'll fix it.

Reply:
What do you think about names like `_0`, `_1`, etc. instead of `item0`, to further differentiated it from regular objects type.

Reply:
As a GraphQL type, having `itemN` properties feels more natural to me (probably influenced by C# tuples and javatuples).